macros/materializations/insert_by_period_materialization.sql

```diff
@@ -84,10 +84,7 @@
 
   -- `begin` happens here, so `commit` after it to finish the transaction
   {{run_hooks(pre_hooks, inside_transaction=True)}}
-  {% call statement() -%}
-    begin; -- make extra sure we've closed out the transaction
-    commit;
-  {%- endcall %}
+  {{ adapter.commit() }}
```

Comment:
this materialization previously used 
```
begin;
commit;
```
to make sure that the following code does _not_ run inside of a transaction. We can't just call `commit` here directly, as the database will return an error if a transaction isn't currently open. I think the previous approach was suboptimal, but I think this change lead to the test failure that you can see in the CircleCI build.

Reply:
It's also worth saying: `adapter.commit()` and `commit;` do slightly different things. Calling `adapter.commit()` will tell dbt that the active connection has a transaction open, whereas running `commit;` explicitly will not. I'm very open to using `adapter.commit()` here, but I do think there's some other work required to make this work correctly in practice.

Reply:
Interesting, I thought that `run_hooks` with `inside_transaction=True` would either begin a transaction or continue an existing one, perhaps it commits or otherwise ends the transaction internally. If there is no transaction at this point then the `{{ adapter.commit() }}` might be redundant.

I've looked a bit more and see now that Redshift and postgres before v11 [don't support nested transactions](https://stackoverflow.com/a/25428060/1335793), and Redshift has no support for savepoint or subtransactions either. I've misunderstood the behaviour. So you could `begin` a bunch of times and only need to `commit` or `rollback` once. The previous code here is therefore not harmless at worst and safe at best - doing what the comment said. I'll revert it.